lib/runLoop.ts

```diff
@@ -45,8 +47,29 @@ const runLoop = async ({
   //   escrowDB, ladderTiers, useTinyMan, api,
   // environment, orderAlgoDepth} = config;
 
+  if (!config.poolInfoAddr && !exitLoop && config.api) {
+    assetInfo = await getAssetInfo({
+      indexerClient: config.api.indexer,
+      assetId: config.assetId,
+    });
+
+    const poolInfo = await getTinymanPoolInfo(
+      config.environment,
+      config.assetId,
+      assetInfo.asset.params.decimals
+    );
+
+    if (poolInfo) {
+      config.poolInfoAddr = poolInfo?.addr;
+    } else {
+      stopLoop({ config });
+      return;
+    }
+  }
+
```

Comment:
I think the run loop input should still be frozen, since it is input. The pool info should be stored in a separate state variable. For example at the beginning of the file

let poolInfo;